sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-integration.service.ts

```diff
@@ -1,31 +1,121 @@
 import { Injectable } from "@nestjs/common";
-import { FixedFormatFileLine } from "src/services/ssh/sftp-integration-base.models";
+import { FixedFormatFileLine } from "../../services/ssh/sftp-integration-base.models";
 import { SFTPIntegrationBase } from "../../services/ssh/sftp-integration-base";
-import { ECertRecord } from "./e-cert-integration-model";
+import { ECertFileFooter } from "./e-cert-files/e-cert-file-footer";
+import { ECertFileHeader } from "./e-cert-files/e-cert-file-header";
+import { ECertResponseRecord } from "./e-cert-files/e-cert-response-record";
+import {
+  ECertRecord,
+  RecordTypeCodes,
+} from "./models/e-cert-integration-model";
+import { OfferingIntensity } from "../../database/entities";
+import { ECertPartTimeResponseRecord } from "./e-cert-part-time-integration/e-cert-files/e-cert-response-record";
+import { ECertFullTimeResponseRecord } from "./e-cert-full-time-integration/e-cert-files/e-cert-response-record";
 
 @Injectable()
-export abstract class ECertIntegrationService<
-  DownloadType,
-> extends SFTPIntegrationBase<DownloadType> {
+export abstract class ECertIntegrationService extends SFTPIntegrationBase<
+  ECertResponseRecord[]
+> {
   /**
    * This method will be implemented in the extended class and is used to create the ECert request content.
    * @param ecertRecords
    * @param fileSequence
    */
   createRequestContent(
     ecertRecords: ECertRecord[],
-    fileSequence: number,
+    fileSequence?: number,
   ): FixedFormatFileLine[] {
     throw new Error(
       `Method not implemented , ${ecertRecords} && ${fileSequence} not declared.`,
```

Comment:
This method seems to be mandatory to every class deriving from `ECertIntegrationService` can we please make it abstract?